astropy/coordinates/erfa_astrom.py

```diff
@@ -0,0 +1,260 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This module contains a helper function to fill erfa.astrom struct and a
+ScienceState, which allows to speed up coordinate transformations at the
+expense of accuracy.
+"""
+import warnings
+
+import numpy as np
+import erfa
+
+from ..time import Time
+from ..utils.state import ScienceState
+from .. import units as u
+from .builtin_frames.utils import (
+    get_jd12, get_cip, prepare_earth_position_vel, get_polar_motion, get_dut1utc,
+    pav2pv
+)
+
+from ..utils.exceptions import AstropyWarning
+
+__all__ = []
+
+
+class ErfaAstrom:
+    '''
+    The default provider for astrometry values.
+    A utility class to extract the necessary arguments for
+    erfa functions from frame attributes, call the corresponding
+    erfa functions and return the astrom object.
+    '''
+
+    @staticmethod
+    def apci(frame):
+        '''
+        Wrapper for ``erfa.apci``, used in conversions CIRS <-> ICRS
+
+        Arguments
+        ---------
+        frame: ``astropy.coordinate.CIRS``
```

Comment:
At least one of these expected a frame attribute that was not present for `AltAz` frames.

Reply:
the `obsgeoloc` and `obsgeovel` frame attributes of the `GCRS` frame are needed in `apcs` and those attributes are not present on an `AltAz` frame.

Reply:
Yes - I see that passing the wrong frame won’t work. 

That wasn’t the point I was trying to make. It’s that your doc string is too specific. The argument is either a frame or a skycoord. The docstring should specify that both are allowed. 

Since there’s no such class as a CIRSSkycoord so the explanation of what frame is expected needs to go below.

Reply:
done